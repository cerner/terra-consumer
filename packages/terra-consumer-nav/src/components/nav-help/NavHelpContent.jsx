import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import Toggler from 'terra-toggle';
import Arrange from 'terra-arrange';
import IconChevronUp from 'terra-icon/lib/icon/IconChevronUp';
import IconChevronDown from 'terra-icon/lib/icon/IconChevronDown';
import navElementShape from '../../NavPropShapes';
import styles from './NavHelp.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * A list of items to be displayed in help menu/popup/modal.
   */
  helpContent: PropTypes.arrayOf(PropTypes.shape(
    navElementShape,
    {
      children: PropTypes.arrayOf(PropTypes.shape(
      navElementShape,
        {
          children: PropTypes.array,
        },
      )),
    },
  )),
};

const defaultProps = {
  helpContent: [],
};

class NavHelpContent extends React.Component {
  constructor() {
    super();
    this.state = ({ isOpen: false });

    this.handleToggle = this.handleToggle.bind(this);
  }

  handleToggle() {
    this.setState({ isOpen: !this.state.isOpen });
  }

  render() {
    const { helpContent, ...customProps } = this.props;
    const toggleIcon = this.state.isOpen ? <IconChevronUp className={cx('icon')} /> : <IconChevronDown className={cx('icon')} />;
    const contentList = helpContent.map((content, i) => {
      const contentElement = content.children.length > 0 ?
      (<button key={`${content.text}`} onClick={this.handleToggle} className={i > 0 ? cx('help-item', 'help-item-border') : cx('help-item')}>
        <div className={cx('help-item-text')}>
          <Arrange
            align="stretch"
            fitStart={<div className={cx('icon')}>{content.icon}</div>}
            fill={<div className={cx('item-text-padding')}>{content.text}</div>}
            fitEnd={<div className={cx('icon')}>{toggleIcon}</div>}
          />
          <Toggler isOpen={this.state.isOpen} isAnimated className={cx('toggler-padding')}>
            { content.children.map(element => (
              <div key={`${element.text}`} className={cx('help-subitem', 'toggler-content-padding')}>{element.text}</div>
              ))
            }
          </Toggler>
        </div>
      </button>
      )
      :
      (
        <button key={`${content.text}`} onClick={() => { window.location = content.uri; }} href="content.uri" className={i > 0 ? cx('help-item', 'help-item-border') : cx('help-item')} >
          <div className={cx('help-item-text')}>
            <Arrange
              align="center"
              fitStart={<svg className={cx('icon')}>{content.icon}</svg>}
              fill={<div className={cx('item-text-padding')}>{content.text}</div>}
            />
          </div>
        </button>
      );
      return contentElement;
    });

    return (
      <div {...customProps}>
        {contentList}
      </div>
    );
  }
}

NavHelpContent.propTypes = propTypes;
NavHelpContent.defaultProps = defaultProps;
export default NavHelpContent;
