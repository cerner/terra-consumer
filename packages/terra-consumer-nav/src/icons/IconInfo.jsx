/* eslint-disable */
import React from 'react';
import IconBase from 'terra-icon/lib/IconBase';

const SvgIcon = (customProps) => {
  const attributes = Object.assign({}, customProps);

  return (
    <IconBase {...attributes}>
      <path d="M24,1C11.317,1,1,11.318,1,24s10.317,23,23,23s23-10.318,23-23S36.683,1,24,1z M24,45C12.421,45,3,35.58,3,24S12.421,3,24,3s21,9.42,21,21S35.579,45,24,45z" />
      <circle cx="24" cy="36" r="2" />
      <path
        d="M24,11c-2.642,0-5.375,1.868-6.5,4.442c-0.222,0.506,0.01,1.096,0.516,1.317c0.508,0.222,1.097-0.01,1.316-0.516
        C20.141,14.395,22.147,13,24,13c2.663,0,5,2.336,5,5s-2.337,5-5,5c-0.553,0-1,0.448-1,1v5c0,0.552,0.447,1,1,1s1-0.448,1-1v-4.075
        c3.341-0.502,6-3.47,6-6.925C31,14.206,27.794,11,24,11z"
      />
    </IconBase>
  );
};

SvgIcon.displayName = "IconInfo";
SvgIcon.defaultProps = {"viewBox":"0 0 48 48","xmlns":"http://www.w3.org/2000/svg","data-name":"Layer 1"};

export default SvgIcon;
/* eslint-enable */
