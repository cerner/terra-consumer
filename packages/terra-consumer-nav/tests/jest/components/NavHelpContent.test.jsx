import React from 'react';
import NavHelpContent from '../../../src/components/nav-help/NavHelpContent';
import IconInfo from '../../../src/icons/IconInfo';

const testData = [
  {
    text: 'Technical Questions',
    uri: 'http://localhost:8083/',
    icon: (<IconInfo />),
    children: [{
      text: 'Need help using this portal or need to report an issue? Contact the support team at 123-xxx-xxxx',
      uri: 'http://localhost:8083/',
    }],
  },
  {
    text: 'Get Support ID',
    uri: 'http://localhost:8083/',
    icon: (<IconInfo />),
    children: [],
  },
];

describe('Content of the Help Modal/Popup', () => {
  // Snapshot Tests
  it('should render a list of items with/without toggler', () => {
    const wrapper = shallow(<NavHelpContent helpContent={testData} />);
    expect(wrapper).toMatchSnapshot();
  });
});
