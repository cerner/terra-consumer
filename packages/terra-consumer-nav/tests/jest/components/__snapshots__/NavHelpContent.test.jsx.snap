// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content of the Help Modal/Popup should render a list of items with/without toggler 1`] = `
<div>
  <button
    className="help-item"
    onClick={[Function]}
  >
    <div
      className="help-item-text"
    >
      <Arrange
        align="stretch"
        fill={
          <div
            className="item-text-padding"
          >
            Technical Questions
          </div>
        }
        fitEnd={
          <div
            className="icon"
          >
            <IconChevronDown
              className="icon"
              data-name="Layer 1"
              isBidi={true}
              viewBox="0 0 48 48"
              xmlns="http://www.w3.org/2000/svg"
            />
          </div>
        }
        fitStart={
          <div
            className="icon"
          >
            <IconInfo
              data-name="Layer 1"
              viewBox="0 0 48 48"
              xmlns="http://www.w3.org/2000/svg"
            />
          </div>
        }
      />
      <Toggle
        className="toggler-padding"
        isAnimated={true}
        isOpen={false}
      >
        <div
          className="help-subitem toggler-content-padding"
        >
          Need help using this portal or need to report an issue? Contact the support team at 123-xxx-xxxx
        </div>
      </Toggle>
    </div>
  </button>
  <button
    className="help-item help-item-border"
    href="content.uri"
    onClick={[Function]}
  >
    <div
      className="help-item-text"
    >
      <Arrange
        align="center"
        fill={
          <div
            className="item-text-padding"
          >
            Get Support ID
          </div>
        }
        fitStart={
          <svg
            className="icon"
          >
            <IconInfo
              data-name="Layer 1"
              viewBox="0 0 48 48"
              xmlns="http://www.w3.org/2000/svg"
            />
          </svg>
        }
      />
    </div>
  </button>
</div>
`;
